### ISSUE TITLE
Parse error when last form is `#_:clj-kondo/ignore`

### ISSUE THREAD

**Issue by colin-p-hill** [clj-kondo/clj-kondo#1953]
**version**

v2023.01.12

**platform**

Windows 11, native, scoop.

**editor**

Cursive, clojure-extras.

**problem**

Clj-kondo reports a parse error when the last form in the file is `#_:clj-kondo/ignore` (or its map syntax). This may happen, for instance, when trying to ignore a linter finding in a comment.

**repro**

```clojure
(ns repro)
#_:clj-kondo-ignore
;; This is a very long comment aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
```
```shell
> clj-kondo --lint repro.clj
repro.clj:0:0: error: Can't parse repro.clj, clojure.lang.PersistentList cannot be cast to clojure.lang.IPersistentMap
linting took 4ms, errors: 1, warnings: 0
```

**config**

```edn
{:linters {:line-length {:max-line-length 80}}}
```

**expected behavior**

I expect either no error, or an error which explains that `#_:clj-kondo/ignore` must precede a form (and not a comment).

**Comment by borkdude** [comment]
`#_:clj-kondo/ignore` is supposed to go before a form. A comment is not a form. I'll take a deeper look at this later
